### ISSUE TITLE
Pretrends naive run

### ISSUE THREAD

**Issue by SundaramReka** [jonathandroth/pretrends#2]
Hi! Jonathan:

I did a naïve run of your code. When running the segment

pretrendsResults <- 
  pretrends(betahat = beta, 
            sigma = sigma, 
            tVec = tVec, 
            referencePeriod = referencePeriod,
            deltatrue = slope50 * (tVec - referencePeriod))

I get the following error:

Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/Library/Frameworks/R.framework/Versions/4.0/Resources/library/gmm/libs/gmm.so':
  dlopen(/Library/Frameworks/R.framework/Versions/4.0/Resources/library/gmm/libs/gmm.so, 6): Library not loaded: /usr/local/gfortran/lib/libgomp.1.dylib
  Referenced from: /Library/Frameworks/R.framework/Versions/4.0/Resources/library/gmm/libs/gmm.so
  Reason: image not found

Note. I just installed and was following along your steps.

best,

reka

**Comment by jonathandroth** [comment]
Hi - thanks!

This sounds like an R package installation issue. Can you remove the
package, restart R, and try again.  If that doesn't work, some of the other
solutions here might help:
https://stackoverflow.com/questions/38943764/r-error-in-dyn-loadfile-dllpath-dllpath

Please let me know if it works out!

J

**Comment by SundaramReka** [comment]
It is struggling with my version of R 4.0.2 and packages cli, vctrs, and zoo. I will look into it later.

**Comment by SundaramReka** [comment]
This is definitely a MAC user issue. I will need to wait until someone better skilled than I can post something about resolving it.

Here is a link to some other package running into a similar problem.
https://github.com/r-spatial/sf/issues/1536#issuecomment-727342736

**Comment by jonathandroth** [comment]
Unfortunately sometimes R packages don't install properly and there's not
much that package creators can do about it. Hopefully one of those links
works for you! Keep me posted :)

**Comment by SundaramReka** [comment]
I gave it time. Reinstalled a new Mac, and choose non option under none for installing supporting packages. Because the stack overflow recommended installing each separately. It didn’t work.

And I tried a couple tricks I know. Something goes wrong in the compiling process. It maybe a MAC issue.

r

On May 4, 2022, at 11:55 AM, Reka Sundaram-Stukel ***@***.******@***.***>> wrote:

Of course, it has to be stupid like that. I will try after work and email you.

r

On May 4, 2022, at 11:54 AM, Jonathan Roth ***@***.******@***.***>> wrote:


Hi - thanks!

This sounds like an R package installation issue. Can you remove the
package, restart R, and try again. If that doesn't work, some of the other
solutions here might help:
https://stackoverflow.com/questions/38943764/r-error-in-dyn-loadfile-dllpath-dllpath

Please let me know if it works out!

J

**Comment by jonathandroth** [comment]
Sorry to hear that you're having trouble. The error message that you sent
appeared to be an issue with the gmm package. Have you tried uninstalling
and reinstalling that package directly? Unfortunately, I don't think
there's much else I can do to help with package installation issues
(particularly for dependent packages like gmm). I've successfully installed
the package on my own Mac computers without a problem. FYI, if you can't
get the R package to work there is also a Shiny app:
https://jonathandroth.shinyapps.io/pretrendspowerapp/

Good luck, and please keep me posted!

**Comment by SundaramReka** [comment]
Will give it a try this weekend and let you know.

Reka

On Oct 12, 2022, at 8:44 AM, Jonathan Roth ***@***.***> wrote:

﻿

Sorry to hear that you're having trouble. The error message that you sent
appeared to be an issue with the gmm package. Have you tried uninstalling
and reinstalling that package directly? Unfortunately, I don't think
there's much else I can do to help with package installation issues
(particularly for dependent packages like gmm). I've successfully installed
the package on my own Mac computers without a problem. FYI, if you can't
get the R package to work there is also a Shiny app:
https://jonathandroth.shinyapps.io/pretrendspowerapp/

Good luck, and please keep me posted!

**Comment by SundaramReka** [comment]
Now I am aware why you sent the email response out of the blue. You are in Madison. I will try to stop over to see your talk.

Because I work for the State of Wisconsin, and my honorary fellow is new—I am not on the email lists for these things. I need to get on them. Being a returning adult means I need to start investigating different opportunities for econometric socialization while I regain footing in academia.


Looking forward.

Best,

reka

**Comment by jonathandroth** [comment]
Thanks for coming!

I'm slightly confused though. I got a notification of your response from
GitHub three days ago. Did you send your reply earlier?

**Comment by SundaramReka** [comment]
Looks like the conference is full so may not be able to attend.
reka

**Comment by jonathandroth** [comment]
Thanks for trying! And that's very weird re the delay in the GitHub
notification. Sorry about that!

**Comment by SundaramReka** [comment]
Yeah. I asked Chris Taber if I could gate crash, he said the administrator was right it was fully booked. I always think these things work out for the best. I didn’t anger State employees by taking the day off without notice, and Chris became aware I need to be included in emails that he is unaware I am not included in.

It is all good. 

I hope you enjoyed Madison, it is a beautiful city. 

Best, reka

**Comment by jonathandroth** [comment]
Got it, well thanks for trying and hope we can connect soon! Madison is
great!

(FYI, this thread is responding to the git issue and thus is public for
anyone to see...)
